File: day_7/caesar_cipher/caesarCipher.js
exports.caesarCipher = function(sen, num) {
  let result = "";
  let alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
  for (let char of sen) {
    if (alphabet.includes(char.toLowerCase())) {
      charIndex = alphabet.findIndex(letter => letter === char.toLowerCase())
      newIndex = (charIndex + num) % 26;
      if (newIndex < 0) {
        newIndex = 26 + newIndex
      }
      if (char === char.toUpperCase()){ 
        char = alphabet[newIndex].toUpperCase()
      } else {
        char = alphabet[newIndex]
      } 
    }
    result += char
  } return result
};

File: day_7/caesar_cipher/caesar_cipher.py
import string

def caesar_cipher(sen, shift_amount):
  alphabet = string.ascii_lowercase
  result = ""
  for char in sen:
    if char.isalpha():
      char_index = alphabet.index(char.lower())
      new_index = (char_index + shift_amount) %26
      if char.isupper():
        char = alphabet[new_index].upper()
      else:
        char = alphabet[new_index]
    result += char
  return result